Support for multiple "On Done Conditions"

dCastanho opened the pull request:
The goal is to have multiple conditions with different error messages.

dCastanho committed:
Support for multiple conditions in Work Queues

dCastanho committed:
Updated definition

dCastanho committed:
Accounted for empty error messages

jbarata commented on integrationm/concurrent/_wrkfl_change-work-item-state.groovy:
```diff
@@ -20,36 +20,52 @@ if (nextState == "Done") {
 
-            if (doneConditions != null) {
+            def conditions = wq.fields
+                    .findAll{f -> f.fieldDefinition.name.startsWith("Condition") }
```
especifica mais um bocadinho o filtro; neste momento não há colisões mas há para lá mais campos com condition no nome e no me gusta.
Sugiro que ponhas o pipe no nome `f.fieldDefinition.name.startsWith("Condition |")`

jbarata replied:
esquece este comentário acima.

Eu e o Miguel estivemos a rever a UI e não gostámos do placeholder no group condition (além disso temos aquele bug de placeholder com duplicáveis ficam todos iguais.

Assim, mudámos a definição para o nome do campo group ficar apenas `Condition` e com estilo standard

<img width="1294" alt="image" src="https://github.com/user-attachments/assets/d899f3d1-2871-4ae7-9249-0037bbcc25b7" />


Depois actualiza apenas o readme e o json se necessário

dCastanho replied:
Ok, ignorado 👍 foi mais ou menos por isso que não pus o placeholder no codigo, caso quisessemos tirar. Vou actualizar o readme e o json.

jbarata commented on integrationm/concurrent/_wrkfl_change-work-item-state.groovy:
```diff
@@ -20,36 +20,52 @@ if (nextState == "Done") {
-            def doneConditionsErrorMsg = wq.value("Done Conditions Error Msg") ?: doneConditions
 
-            if (doneConditions != null) {
+            def conditions = wq.fields
```
mudaste o nome para apenas conditions e não acho bem; há para lá outros tipos de conditions e fica menos explicito que são condições de done.
Podes re-aproveitar o nome `doneConditions`; embora agora não seja tecnicamente o mesmo que era antes, na prática é o que é!

jbarata reviewed (changes requested):
2 comentáriozitos

dCastanho committed:
Renamed fields for clarity

dCastanho committed:
Better error handling

When there was an error evaluating a condition, instead of return, it was continuing to evaluate the other conditions normally - but setting the state to Error.

dCastanho committed:
Updated definition

jbarata commented on others/customize.workflows/README.md:
````diff
@@ -51,6 +51,21 @@ instances:update:<def_id_work_item>:
+```
+NOVOS:
+(After "On Done")
+Work Specification > Possible States >  Condition : $group $style[subgrupo] (Duplicable)
````
isto já não é $style[subgrupo] e tem um expanded

dCastanho committed:
Updated readme

jbarata reviewed (approved):
LGTM
